From cd36cfa5dae6365c67d8f51d7cac74698f017331 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Mon, 24 Feb 2025 20:37:51 +0100
Subject: [PATCH 1304/1695] dts: stm32: fix GPU OPP description for STM32MP23x

Since commit e80e43fed285d ("plat-stm32mp: introduce GPU OPP driver") the
GPU opp are described in device tree and are used for SCMI power domain.

But the OPP for STM32MP23 SoC is correctly defined as it is limited at
400MHz.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git dc74bac3cbb271e524cc55a8319327b59fa65e9c dts: stm32: fix GPU OPP description for STM32MP23x]
Fixes: 7f7186808936 ("dts: stm32: add GPU OPP description in stm32mp235.dtsi")
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Ibd65f24a874392fa25e44d9149356535dbd582ab
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/439608
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp235.dtsi | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp235.dtsi b/core/arch/arm/dts/stm32mp235.dtsi
index 87a51b1ef..7c4dd7ace 100644
--- a/core/arch/arm/dts/stm32mp235.dtsi
+++ b/core/arch/arm/dts/stm32mp235.dtsi
@@ -10,19 +10,12 @@
 	gpu_opp_table: opp-table-gpu {
 		compatible = "operating-points-v2";
 
-		opp-800000000 {
-			opp-hz = /bits/ 64 <800000000>;
+		opp-400000000 {
+			opp-hz = /bits/ 64 <400000000>;
 			opp-microvolt = <800000>;
 			opp-supported-hw = <0x3>;
 			st,opp-default;
 		};
-
-		opp-900000000 {
-			opp-hz = /bits/ 64 <900000000>;
-			opp-microvolt = <900000>;
-			opp-supported-hw = <0x2>;
-			st,opp-default;
-		};
 	};
 };
 
-- 
2.47.2

